x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/CertificateTool.java

```diff
@@ -924,16 +929,43 @@ static PEMEncryptor getEncrypter(char[] password) {
         return new JcePEMEncryptorBuilder("AES-128-CBC").setProvider(BC_PROV).build(password);
     }
 
-    private static <T, E extends Exception> T withPassword(String description, char[] password, Terminal terminal,
+    static boolean checkPasswordLengthForOpenSSLCompatibility(char[] password, Terminal terminal, boolean confirm) {
```

Comment:
Nit:
This method and parameter names confused me for a bit. They made me think that the behaviour when `confirm` is `false` should be `return false` if password is too long. But instead the default behaviour is `return true` unless user says otherwise and `confirm` is to let user have that say. Maybe:
```suggestion
    static boolean maybeWarnAndConfirmPasswordLengthForOpenSSLCompatibility(char[] password, Terminal terminal, boolean confirmRequired) {
```
It is quite verbose, but it feels more accurate to me. I am also happy if you can come up with something better.